libs/community/langchain_community/vectorstores/azure_cosmos_db.py

```diff
@@ -409,17 +538,93 @@ def max_marginal_relevance_search_by_vector(
         return mmr_docs
 
     def max_marginal_relevance_search(
-        self,
-        query: str,
-        k: int = 4,
-        fetch_k: int = 20,
-        lambda_mult: float = 0.5,
-        **kwargs: Any,
+            self,
+            query: str,
+            k: int = 4,
+            kind: CosmosDBVectorSearchType = CosmosDBVectorSearchType.VECTOR_IVF,
+            fetch_k: int = 20,
+            lambda_mult: float = 0.5,
+            score_threshold: Optional[float] = None,
+            **kwargs: Any,
     ) -> List[Document]:
         # compute the embeddings vector from the query string
         embeddings = self._embedding.embed_query(query)
 
         docs = self.max_marginal_relevance_search_by_vector(
-            embeddings, k=k, fetch_k=fetch_k, lambda_mult=lambda_mult
+            embeddings, k=k, kind=kind, fetch_k=fetch_k, lambda_mult=lambda_mult, score_threshold=score_threshold
         )
         return docs
+
+    def get_collection(self):
+        return self._collection
+
+
+class AzureCosmosDBVectorSearchRetriever(BaseRetriever):
```

Comment:
how's this class different from VectorStoreRetriever?

Reply:
I have removed the AzureCosmosDBVectorSearchRetriever class, as it is identical implementation to the VectorStoreRetriever class.